R/utils.R

```diff
@@ -208,64 +208,20 @@ cpp_options_to_compile_flags <- function(cpp_options) {
   paste0(cpp_built_options, collapse = " ")
 }
 
-prepare_precompiled <- function(cpp_options = list(), quiet = FALSE) {
-  flags <- NULL
-  if (!is.null(cpp_options$stan_threads)) {
-    flags <- c(flags, "threads")
-  }
-  if (!is.null(cpp_options$stan_mpi)) {
-    flags <- c(flags, "mpi")
-  }
-  if (!is.null(cpp_options$stan_opencl)) {
-    flags <- c(flags, "opencl")
-  }
-  if (is.null(flags)) {
-    flags <- "noflags"
-  } else {
-    flags <- paste0(flags, collapse = "_")
-  }
-  main_path_w_flags <- file.path(cmdstan_path(), "src", "cmdstan", paste0("main_", flags, ".o"))
-  main_path_o <- file.path(cmdstan_path(), "src", "cmdstan", "main.o")
-  model_header_path_w_flags <- file.path(cmdstan_path(), "stan", "src", "stan", "model", paste0("model_header_", flags, ".hpp.gch"))
-  model_header_path_gch <- file.path(cmdstan_path(), "stan", "src", "stan", "model", "model_header.hpp.gch")
-  if (file.exists(model_header_path_gch)) {
-    model_header_gch_used <- TRUE
-  } else {
-    model_header_gch_used <- FALSE
-  }
-  if (!file.exists(main_path_w_flags)) {
-    message(
-      "Compiling the main object file and precompiled headers (may take up to a few minutes). ",
-      "This is only necessary the first time a model is compiled after installation or when ",
-      "threading, MPI or OpenCL are used for the first time."
-    )
-    clean_compile_helper_files()
-    run_log <- processx::run(
-      command = make_cmd(),
-      args = c(cpp_options_to_compile_flags(cpp_options),
-               main_path_o),
-      wd = cmdstan_path(),
-      echo_cmd = !quiet,
-      echo = !quiet,
-      spinner = quiet,
-      stderr_line_callback = function(x,p) { if (!quiet) message(x) },
-      error_on_status = TRUE
-    )
-    file.copy(main_path_o, main_path_w_flags)
-    if (model_header_gch_used) {
-      run_log <- processx::run(
-        command = make_cmd(),
-        args = c(cpp_options_to_compile_flags(cpp_options),
-                 file.path("stan", "src", "stan", "model", "model_header.hpp.gch")),
-        wd = cmdstan_path(),
-        echo_cmd = !quiet,
-        echo = !quiet,
-        spinner = quiet,
-        stderr_line_callback = function(x,p) { if (!quiet) message(x) },
-        error_on_status = TRUE
-      )
-      file.copy(model_header_path_gch, model_header_path_w_flags)
+check_stanc_options <- function(stanc_options) {
```

Comment:
Is the `i` (and the `i <- i + 1` below) used in this function necessary? Can't we just replace all `i` with the `s` that's in the for loop? Or am I missing something (totally possible)?

Reply:
Hm, I need the `i` for the names. Not sure how else I can get the name from the list value? I am still an R noob.

Reply:
I might be misunderstanding but aren't `i` and `s` always the same value in this for loop? If so then all `i` could be replaced with `s`. But maybe I'm being an idiot ;)

Reply:
Basically for 
```
list(
"test",
"test1" = TRUE
)
```
s would be
test and TRUE but I need the name in the second case.

Reply:
Oh I'm an idiot. `s` isn't an integer at all it's the stanc option!

Reply:
So just ignore this, what you have is fine!

Reply:
Yeah I was reading it too quickly and assuming s was an index but it's the option itself.